src/main/scala/io/archivesunleashed/app/WriteGraphML.scala

```diff
@@ -46,7 +55,7 @@ object WriteGraphML {
   def makeFile (rdd: RDD[((String, String, String), Int)], graphmlPath: String): Boolean = {
     val outFile = Files.newBufferedWriter(Paths.get(graphmlPath), StandardCharsets.UTF_8)
     val edges = rdd.map(r => "<edge source=\"" + ComputeMD5RDD(r._1._2.getBytes) + "\" target=\"" +
-      ComputeMD5RDD(r._1._3.getBytes) + "\"  type=\"directed\">\n" +
+      ComputeMD5RDD(r._1._3.getBytes) + "\" type=\"directed\">\n" +
```

Comment:
Using a hash instead of a unique id resulted in lost data in the past due to overloading. To fix, I believe you can use the monotonicallyIncreasingId() function as per: https://stackoverflow.com/a/36946952/3050664.  There is another solution that involves moving from rdd to df(schema). I am not sure of the extent that adding the schema can prevent past performance issues we had moving from rdd to df.

Reply:
Hi Ryan – this'd be out of scope on this PR. If you think it's significant, do you want to add an AUT issue?